lib/crowdin/adapter.rb

```diff
@@ -94,18 +94,21 @@ def upload_attributes_to_translate(object_type, object_id, updated_at, attribute
     # Failures can be:
     # 1) +CrowdIn::Client::Errors::Error+ if we fail to fetch the translation status of files
     # 2) +CrowdIn::FileMethods::FilesError+ if any individual files fail on export
-    def translations(language)
+    def translations(language, batch_size = 500)
       begin
-        file_status_list = @client.language_status(language)
-        file_ids = file_status_list.map { |f| f['file_id'] }
-        translations_for_files(file_ids, language)
+        # Get all file ids at once, since that is a single API call that is cached
+        # but fetch translations in batches and yield them to the caller.
+        file_ids = @client.files.map { |f| f['id'] }
+        file_ids.each_slice(batch_size) do |batch|
+          yield(translations_for_files(batch, language))
+        end
       rescue CrowdIn::Client::Errors::Error => e
```

Comment:
would it be helpful to log which file_ids failed?

Reply:
In this case, the failure could come from the `files` call, or for the `translations_for_files` call. My hope is that the place producing the error would log that correctly, which is what I'm trying to work towards.